### PR TITLE
Canpass refactor

### PR DESCRIPTION
**Pull request by Rohesie**
Porting https://github.com/tgstation/tgstation/pull/48659
Might want to test it a round or two for issues.

### CODE REVIEW

File: code/__DEFINES/flags.dm [hunk 1]
```diff
@@ -15,3 +15,11 @@ GLOBAL_LIST_INIT(bitflags, list(1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 204
 #define DF_USE_TAG		(1<<0)
 #define DF_VAR_EDITED	(1<<1)
 #define DF_ISPROCESSING (1<<2)
+
+//Movement Types
+#define GROUND			(1<<0)
+#define FLYING			(1<<1)
+#define VENTCRAWLING	(1<<2)
+#define FLOATING		(1<<3)
```
Comment (psykzz) [inline code/__DEFINES/flags.dm:23]: How is floating different from flying?

Comment (Rohesie) [reply code/__DEFINES/flags.dm:23]: Floating is from lack of gravity. Flying is self-powered floating with more control, I guess.

Comment (Rohesie) [reply code/__DEFINES/flags.dm:23]: I guess flying is also higher? Like, you won't fall over a railing if you are floating in no gravity unless you purposely jump over, but you can fly over it.
The difference is not huge and clearly defined, granted, as z-levels and stuff are not yet very well consolidated. But this is how /tg/ does it currently.

Comment (psykzz) [reply code/__DEFINES/flags.dm:23]: Im just looking to get them defined so we know how we want to apply them.,

For example i'd see floating as no higher than normal.
They would never slip or touch objects on the floor, but they wouldn't float above things like windows / tables etc

Comment (Rohesie) [reply code/__DEFINES/flags.dm:23]: Sounds fair to me.

**Comment by psykzz** [comment]
Tested this for a round today.
Marines on help intent didn't swap places with other marines.

```
#3936: 'Chem rework mk3 the nerfening' by jroinc at commit 2357ce2552
#3998: 'autoshotgun changes' by Pariah919 at commit a54543ca6a
#4014: 'Better flamer refueling methods and a standard flamer buff' by XElectricX at commit b7485566db
#4038: 'Canpass refactor' by Rohesie at commit 5956ddce98
#4048: 'Unpowered Doors Opened by Xenos Quickly' by BobSlim at commit ecb46c0cc3
#4050: 'Queen no longer scales horrifically badly. Has slightly less peak melee damage. Spits faster.' by Pariah919 at commit 8940fb9df8
#4054: 'spitters and prae spit faster' by Pariah919 at commit cbff49c93e
#4059: 'Soft/Hard armor generalized for objects and turfs' by Rohesie at commit 4f225dcc3f
#4061: 'Limb flags fix' by Rohesie at commit 4fdc8282f3
#4063: 'Crusher charge does 25 *1/2/3 instead of 10 *1/2/3' by Pariah919 at commit c99e68bc67
#4062: 'Smartgunner fires slower and harness is removed. Sunder is a bit more.' by Pariah919 at commit 52875ed2e8
#4024: 'should fix shuttle explosions' by spookydonut at commit e05782c949
#3912: '[DO NOT REMOVE THIS FROM TEST MERGES] Add some debugging info for shuttles' by spookydonut at commit c2400e24d4
#3933: 'United Space Lepidoptera' by Sibg at commit f0be01ca74
#4028: 'Fixes explosions happening at the same time as crash' by Rohesie at commit 8c868aadc7
#4006: 'Zoom fix' by GLACoding at commit a968a2e371
#4037: 'removes cooldown from OB' by BobSlim at commit 0c8841653d
#4046: '"Shotgun Shell Pouch/Belt" can be refilled with any magazines (with loose ammo)' by BobSlim at commit 1295ca2ee2
#4047: 'you can refill mags with other mags' by BobSlim at commit 7a94343e78
#4055: 'Panther Buff Gang' by SeniorScore at commit 25b6505a51
#4065: 'Reqs TGUI' by spookydonut at commit 37994a04cf

Local commit: f2cfde1873ed1d3281441d4bc5e18590a869cbc9
Server tools version: /tg/station 13 Server v3.2.6.2
```

**Pull request closed without merging**